prody/proteins/interactions.py

```diff
@@ -0,0 +1,1109 @@
+    
+    return AllInteractions
+    
+
+def calcInteractionMatrix(atoms):
+    """Calculate matrix with protein interactions which is scored by as follows:
+        (1) Hydrogen bonds +2
+        (2) Salt Bridges +3 (Salt bridges might be included in hydrogen bonds)
+        (3) Repulsive Ionic Bonding -1 
+        (4) Pi stacking interactions +3
+        (5) Pi-cation interactions +3
+        (6) Hydrophobic interactions +1
+       
+    :arg atoms: an Atomic object from which residues are selected
+    :type atoms: :class:`.Atomic`
+    
+    Results can be displayed:
+        >> plt.spy(calcInteractionMatrix(atoms), origin='lower', markersize=1.5)
+        or
+        >> showAtomicMatrix(calcInteractionMatrix(atoms), atoms=atoms.select('name CA'), 
+            interpolation=None, cmap='seismic'); plt.clim([-3,3])"""
+            
+    try:
+        coords = (atoms._getCoords() if hasattr(atoms, '_getCoords') else
+                    atoms.getCoords())
+    except AttributeError:
+        try:
+            checkCoords(coords)
+        except TypeError:
+            raise TypeError('coords must be an object '
+                            'with `getCoords` method')
+    
+    LOGGER.info('Calculating all interactions')
+    AllInteractions = calcProteinInteractions(atoms)
+    InteractionsMap = np.zeros([atoms.select('name CA').numAtoms(),atoms.select('name CA').numAtoms()])
+    resIDs = list(atoms.select('name CA').getResnums())
+    resChIDs = list(atoms.select('name CA').getChids())
+    resIDs_with_resChIDs = list(zip(resIDs, resChIDs))
+    scoring = [2, 3, -1, 3, 3, 1]  #HBs_calculations, SBs_calculations, SameChargeResidues, Pi_stacking, Pi_cation, Hydroph_calculations
+    
+    for nr_i,i in enumerate(AllInteractions):
+        if i != []:
+            for ii in i: 
+                m1 = resIDs_with_resChIDs.index((int(ii[0][3:]),ii[2]))
+                m2 = resIDs_with_resChIDs.index((int(ii[3][3:]),ii[5]))
+                InteractionsMap[m1][m2] = InteractionsMap[m1][m2] + scoring[nr_i]
+  
+    return InteractionsMap
+
+
+def calcLigandInteractions(atoms, select='all not (water or protein or ion)', ignore_ligs=['NAG','BMA','MAN']):
+    """Provide ligand interactions with other elements of the system including protein, water and ions.
+    Results are computed by PLIP program which should be installed in local computer.
+    Note that PLIP will not recognize ligand unless it will be HETATM in PDB file.
+    
+    :arg atoms: an Atomic object from which residues are selected
+    :type atoms: :class:`.Atomic`
+    
+    :arg select: a selection string for residues of interest
+            default is 'all not (water or protein or ion)'
+    :type select: str
+    
+    :arg ignore_ligs: List of ligands which will be excluded from the analysis.
+    :type ignore_ligs: List of str
+    
+    To display results as a list of interactions use showLigandInteractions()
```

Comment:
I'm not sure show is the right starting word for getting a list. It suggests something more visual. You could make this `listLigandInteractions`. This could also be better as a private function that gets called by `calcLigandInteractions` based on a keyword argument such as `output_format`.